v-next/hardhat/src/internal/builtin-plugins/network-manager/type-validation.ts

```diff
@@ -221,54 +220,102 @@ const baseNetworkUserConfigSchema = z.discriminatedUnion("type", [
 ]);
 
 function refineEdrNetworkUserConfig(
-  networkConfig: z.infer<typeof baseNetworkUserConfigSchema>,
+  networkConfig: NetworkUserConfig,
   ctx: RefinementCtx,
 ): void {
   if (networkConfig.type === "edr") {
     const {
-      hardfork = LATEST_HARDFORK,
+      chainType = GENERIC_CHAIN_TYPE,
+      hardfork,
+      chains,
       minGasPrice,
       initialBaseFeePerGas,
       enableTransientStorage,
     } = networkConfig;
 
-    if (hardforkGte(hardfork, HardforkName.LONDON)) {
-      if (minGasPrice !== undefined) {
+    if (hardfork !== undefined && !isValidHardforkName(hardfork, chainType)) {
+      ctx.addIssue({
+        code: z.ZodIssueCode.custom,
+        path: ["hardfork"],
+        message: `Invalid hardfork name ${hardfork} for chainType ${chainType}. Expected ${getHardforks(
```

Comment:
I think we should decide on a consistent way to wrap values (single quotes, double quotes, etc.) and normalize all validation and error messages. I'm pretty sure we already have this tracked somewhere.